source/Halibut/Transport/ActiveTcpConnectionsLimiter.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Halibut.Diagnostics;
+using Halibut.Exceptions;
+
+namespace Halibut.Transport
+{
+    public interface IActiveTcpConnectionsLimiter
+    {
+        IDisposable LeaseActiveTcpConnection(Uri subscriptionId);
+
+        IDisposable CreateUnlimitedLease();
+    }
+
+    public class ActiveTcpConnectionsLimiter : IActiveTcpConnectionsLimiter
+    {
+        readonly HalibutTimeoutsAndLimits timeoutsAndLimits;
+
+        Dictionary<Uri, StrongBox<int>> activeConnectionCountPerSubscriptionId = new();
```

Comment:
I use the `StrongBox<int>` so I don't have to update the value in the dictionary after incrementing it. I can just get this type, then increment it's value